### PR TITLE
feat: expose some Fluent Bit logs in the agent

### PR DESCRIPTION
**Pull request by dzhu**
## Description

If the agent's subordinate Fluent Bit container has an issue, it can be
awkward to debug, since its logs are not exposed in our product and can
even vanish entirely if the container exits. Here, we make the agent (1)
print out any error or warning lines it finds in Fluent Bit's output and
(2) dump all recent logs if the container exits.

## Test Plan

- [x] make Fluent Bit fail in various ways (unable to bind port, unable to connect to master, externally killed) and see that useful logs show up in the agent output

## Commentary (optional)

Perhaps we should revisit using the existing `dockerActor` to manage the Fluent Bit container. I avoided that initially because I decided there was enough difference in what this was doing to not make it worth it, but this one is doing more stuff now.

### CODE REVIEW

File: agent/internal/fluent.go [hunk 1]
```diff
@@ -445,3 +469,17 @@ func (f *fluentActor) Receive(ctx *actor.Context) error {
...
+	if err := trackLogs(ctx, f.docker, f.containerID, ctx.Self()); err != nil {
+		ctx.Log().Errorf("error tracking Fluent Bit logs: %s", err)
+	}
+	ctx.Log().Error("Fluent Bit failed, dumping all recent logs")
+	i0 := f.fluentLogsCount - len(f.fluentLogs)
+	if i0 < 0 {
+		i0 = 0
+	}
+	for i := i0; i < f.fluentLogsCount; i++ {
+		ctx.Log().Error(f.fluentLogs[i%len(f.fluentLogs)].Value)
```
Comment (stoksc) [inline agent/internal/fluent.go:483]: How do you feel about logging this all in one go to maybe make it more readable if agent logs get interleaved?

**Review by stoksc — APPROVED** [review]
lgtm. nice getting this up so fast.

File: agent/internal/fluent.go (hunk 1 above)
Comment (dzhu) [reply agent/internal/fluent.go:483]: These will all come out pretty much instantaneously and I'd rather try not to mess with the line-based output of the agent's own logs, so I'll keep it this way.

Comment (stoksc) [reply agent/internal/fluent.go:483]: Makes sense to me.

**Pull request merged**